src/sentry/static/sentry/app/api.tsx

```diff
@@ -383,20 +395,74 @@ export class Client {
         true
       )(jqXHR, textStatus);
 
-    const xhrRequest = jQuery.ajax({
-      url: fullUrl,
+    const aborter = new AbortController();
+
+    // GET requests may not have a body
+    const body = method !== 'GET' ? data : undefined;
+
+    const headers = new Headers({
+      Accept: 'application/json; charset=utf-8',
+      'Content-Type': 'application/json',
+      // TODO: This is not good enough, see ajaxCsrfSetup for what we really
+      // need to do
+      'X-CSRFToken': getCookie(CSRF_COOKIE_NAME) ?? '',
+    });
+
+    const fetchRequest = fetch(fullUrl, {
       method,
-      data,
-      contentType: 'application/json',
-      headers: {
-        Accept: 'application/json; charset=utf-8',
-      },
-      success: successHandler,
-      error: errorHandler,
-      complete: completeHandler,
+      body,
+      headers,
+      credentials: 'same-origin',
+      signal: aborter.signal,
     });
 
-    const request = new Request(xhrRequest);
+    // XXX(epurkhiser): We're migrating off of jquery, so for now we have a
+    // compatability layer which mimics that of the jquery response objects.
+    fetchRequest
+      .then(async response => {
+        let responseJSON: any;
+        let responseText: any;
+
+        // Try to get JSON out of the response no matter the status
+        try {
+          responseJSON = await response.json();
+        } catch {
+          // No json came out.. too bad
+        }
+
+        // Try to get text out of the response no matter the status
+        try {
+          responseText = await response.text();
+        } catch {
+          // No text came out.. too bad
+        }
+
+        const {ok, status, statusText} = response;
```

Comment:
@EvanPurkhiser what happens if we get no `json` or no `text`? I feel like that should probably be an error, yea?

Reply:
I believe we have a case where responses return [204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) which results in empty for both

Reply:
Also, it looks like jQuery is actually looking at the `Content-Type` of the response to determine what to do with the response. Not sure if we should maintain that behavior here.

Reply:
We passed `Accepts: applications/json`, so afaict we were only ever getting json, so we should be OK I think